js/core/utils/deferred.d.ts

```diff
@@ -3,18 +3,18 @@ interface Callback<T> {
 }
 declare class DeferredObj<T> {
     constructor();
-    state(): string;
+    state(): 'pending' | 'rejected' | 'resolved';
     always(alwaysCallback?: Callback<T>): DeferredObj<T>;
     catch(catchCallback?: Callback<T>): DeferredObj<T>;
     then(resolveCallback?: Callback<T>, rejectCallback?: Callback<T>): DeferredObj<T>;
     done(doneCallback?: Callback<T>): DeferredObj<T>;
     fail(failCallback?: Callback<T>): DeferredObj<T>;
     progress(progressCallback?: Callback<T>): DeferredObj<T>;
-    notify(value?: T, ...args: T[]): DeferredObj<T>;
+    notify(...args: T[]): DeferredObj<T>;
     notifyWith(context: DeferredObj<T>, args?: T[]): DeferredObj<T>;
```

Comment:
I think that we can make this definition event better if we write it in the same manner as we do for the `notify` method:
`notifyWith(context: DeferredObj<T>, ...args: T[]): DeferredObj<T>;`